test/api/controllers/groupController.spec.ts

```diff
@@ -0,0 +1,208 @@
+import sinon from 'sinon';
+import { RequestWithGroup } from '../../../src/types/requests';
+import { NextFunction, Response } from 'express';
+import { Group } from '../../../src/types/group';
+import { UserModel } from '../../../src/data-access/models/userModel';
+import { UserGroupRepository } from '../../../src/data-access/repositories/userGroupRepository';
+import { UserGroupModel } from '../../../src/data-access/models/userGroupModel';
+import { GroupService } from '../../../src/services/groupService';
+import { GroupModel } from '../../../src/data-access/models/groupModel';
+import { GroupDataMapper } from '../../../src/data-access/mappers/groupDataMapper';
+import { UserRepository } from '../../../src/data-access/repositories/userRepository';
+import { GroupController } from '../../../src/api/controllers/groupController';
+import { EntityNotFoundError } from '../../../src/core/errors/entityNotFoundError';
+import { InitializeSequelize } from '../../../src/database/postgreSQL/initializeSequelize';
+
+describe('Group controller tests', () => {
+    let mockReq: RequestWithGroup;
+    let mockRes: Response;
+    let mockNext: NextFunction;
+    let userGroupRepository: UserGroupRepository;
+    let groupService: GroupService;
+    let groupController: GroupController;
+    const mockGroups: Array<Group> = [
```

Comment:
mocks can be moved to separate module and import in the required place. No need to take the list of mocked data within the direct unit test